tests/test_spectral_profile.py

```diff
@@ -0,0 +1,303 @@
+        result: List[float],
+    ):
+        """
+        Test the `generate_from_projected_dos` method.
+        """
+        # Get Outputs and ElectronicDensityOfStates from the simulation fixture
+        outputs = simulation_electronic_dos.outputs[0]
+        electronic_dos = outputs.electronic_dos[0]
+
+        # Add `value`
+        if value is not None:
+            electronic_dos.value = value * ureg('1/joule')
+
+        val = electronic_dos.generate_from_projected_dos(logger)
+        assert (val.magnitude == result).all()
+
+        # Testing orbital + atom projected DOS (3 orbitals + 2 atoms PDOS)
+        assert len(electronic_dos.projected_dos) == 5
+        orbital_projected = electronic_dos.extract_projected_dos('orbital', logger)
+        atom_projected = electronic_dos.extract_projected_dos('atom', logger)
+        assert len(orbital_projected) == 3 and len(atom_projected) == 2
+        atom_projected_names = [atom_pdos.name for atom_pdos in atom_projected]
+        assert atom_projected_names == ['atom Ga', 'atom As']
+        assert (
+            atom_projected[0].entity_ref
+            == outputs.model_system_ref.cell[0].atoms_state[0]
+        )  # `Ga` atom
+        assert (
+            atom_projected[1].entity_ref
+            == outputs.model_system_ref.cell[0].atoms_state[1]
+        )  # `As` atom
+
+    def test_normalize(self):
+        """
+        Test the `normalize` method.
+        """
+        # ! add test when `ElectronicEigenvalues` is implemented
+        pass
+
+
+class TestXASSpectra:
+    """
+    Test the `XASSpectra` class defined in `properties/spectral_profile.py`.
+    """
+
+    # ! Include this initial `test_default_quantities` method when testing your PhysicalProperty classes
+    def test_default_quantities(self):
+        """
+        Test the default quantities assigned when creating an instance of the `XASSpectra` class.
+        """
+        xas_spectra = XASSpectra()
+        assert xas_spectra.iri is None  # Add iri when available
+        assert xas_spectra.name == 'XASSpectra'
+        assert xas_spectra.rank == []
+
+    @pytest.mark.parametrize(
+        'xanes_energies, exafs_energies, xas_values',
+        [
+            (None, None, None),
+            ([0, 1, 2], None, None),
+            (None, [3, 4, 5], None),
+            ([0, 1, 2], [3, 4, 5], [0.5, 0.1, 0.3, 0.2, 0.4, 0.6]),
+        ],
+    )
+    def test_generate_from_contributions(
+        self,
+        xanes_energies: Optional[List[float]],
+        exafs_energies: Optional[List[float]],
+        xas_values: Optional[List[float]],
+    ):
+        """
+        Test the `generate_from_contributions` method.
+        """
+        # ! extend this test
+        xas_spectra = XASSpectra()
+        if xanes_energies is not None:
+            xanes_spectra = SpectralProfile()
+            xanes_spectra.variables = [Energy(grid_points=xanes_energies * ureg.joule)]
+            xanes_spectra.value = [0.5, 0.1, 0.3]
```

Comment:
Woudln't this also be more appropriate as an input parameter?

Reply:
Well, for this testing it is more interesting to keep the Energy axis properly defined. Whatever is in value does not matter too much.